pkg/apis/resource/validation/validation.go

```diff
@@ -372,7 +383,7 @@ func validateResourceClaimStatusUpdate(status, oldStatus *resource.ResourceClaim
 		func(consumer resource.ResourceClaimConsumerReference) (types.UID, string) { return consumer.UID, "uid" },
 		fldPath.Child("reservedFor"))...)
 
-	var allocatedDevices sets.Set[structured.DeviceID]
+	var allocatedDevices sets.Set[structured.SharedDeviceID]
```

Comment:
should there be format validation of the ShareID field if non-nil to ensure it's a valid uid?